india_compliance/gst_india/doctype/gstr_1_beta/gstr_1_beta.py

```diff
@@ -156,12 +167,174 @@ def on_generate(self, data, filters=None):
+    if force:
+        return_log = frappe.get_doc("GST Return Log", log_name)
+        verify_request_in_progress(return_log, force)
+
+    frappe.db.set_value("GST Return Log", log_name, "filing_status", "Not Filed")
+
+
+@frappe.whitelist()
+def get_journal_entries(month_or_quarter, year, company):
+    frappe.has_permission("Journal Entry", "read", throw=True)
+
+    from_date, to_date = get_gstr_1_from_and_to_date(month_or_quarter, year)
+
+    journal_entry = frappe.qb.DocType("Journal Entry")
+    journal_entry_account = frappe.qb.DocType("Journal Entry Account")
+
+    gst_accounts = list(
+        get_gst_accounts_by_type(company, "Sales Reverse Charge", throw=False).values()
+    )
+
+    if not gst_accounts:
+        return
+
+    sales_reverse_charge = (
+        frappe.qb.from_(journal_entry)
+        .join(journal_entry_account)
+        .on(journal_entry.name == journal_entry_account.parent)
+        .select(
+            journal_entry.name,
+        )
+        .where(journal_entry.posting_date.between(getdate(from_date), getdate(to_date)))
+        .where(journal_entry_account.account.isin(gst_accounts))
+        .where(journal_entry.docstatus == 1)
+        .run()
+    )
+
+    if not sales_reverse_charge:
+        return
+
+    sales_invoice = frappe.qb.DocType("Sales Invoice")
+    sales_invoice_taxes = frappe.qb.DocType("Sales Taxes and Charges")
+
+    data = (
+        frappe.qb.from_(sales_invoice)
+        .join(sales_invoice_taxes)
+        .on(sales_invoice.name == sales_invoice_taxes.parent)
+        .select(
+            sales_invoice_taxes.account_head.as_("account"),
+            ConstantColumn("Sales Invoice").as_("reference_type"),
+            Case()
+            .when(
+                sales_invoice_taxes.tax_amount > 0, Sum(sales_invoice_taxes.tax_amount)
+            )
+            .as_("debit_in_account_currency"),
+            Case()
+            .when(
+                sales_invoice_taxes.tax_amount < 0,
+                Sum(sales_invoice_taxes.tax_amount * (-1)),
+            )
+            .as_("credit_in_account_currency"),
+        )
+        .where(sales_invoice.is_reverse_charge == 1)
+        .where(
+            Date(sales_invoice.posting_date).between(
+                getdate(from_date), getdate(to_date)
+            )
+        )
+        .where(IfNull(sales_invoice_taxes.gst_tax_type, "") != "")
+        .where(sales_invoice.docstatus == 1)
+        .groupby(sales_invoice_taxes.account_head)
+        .run(as_dict=True)
+    )
+
+    return data
+
+
+@frappe.whitelist()
+def make_journal_entry(
+    company, company_gstin, month_or_quarter, year, je_accounts_details, auto_submit
+):
+    frappe.has_permission("Journal Entry", "write", throw=True)
+
+    from_date, to_date = get_gstr_1_from_and_to_date(month_or_quarter, year)
+
+    journal_entry = frappe.get_doc(
+        {
+            "doctype": "Journal Entry",
+            "company": company,
+            "company_gstin": company_gstin,
```

Comment:
Remarks: Automated entry on GSTR-1 Filing with ARN and date of filing reference from GST Return log.